HOWTO-publish-a-release.md

```diff
@@ -0,0 +1,301 @@
+
+You must have access to the following repositories:
+
+-   `git@github.openssl.org:omc/data.git`
+
+    This contains files to be updated as part of any release.
+
+-   Any of:
+
+    -   `git@github.openssl.org:openssl/openssl.git`
+
+        This is the public source repository, so is only necessary to stage
+        a public release, which are those that haven't reached End-Of-Life
+        yet.
+
+    -   `git@github.openssl.org:openssl/security.git`
+
+        This is the security source repository, where security fixes are
+        staged before being publically released.  It is used as source
+        repository instead of `openssl/openssl` to stage a security
+        release.
+
+    -   `git@github.openssl.org:openssl/premium.git`
+
+        This is the source repository for premium customers, used for both
+        security and non-security releases.
+
+## SSH access
+
+To perform a release, you must have appropriate access to OpenSSL's
+development host, dev.openssl.org.  To test this, try to log in with ssh:
+
+    ssh dev.openssl.org
+
+You must also check that you can perform tasks as the user 'openssl' on
+dev.openssl.org.  When you have successfully logged in, test your access to
+that user with sudo:
+
+    sudo -u openssl id
+
+# Publish the release
+
+## on SFTP
+
+*BE CAREFUL*  This section makes everything visible and is therefore largely
+irreversible.  If you are performing a dry run then DO NOT perform any steps
+in this section.
+
+*NOTE*  This section should only be performed for public releases, i.e.
+releases made from `git@github.openssl.org:openssl/openssl.git` or
+`git@github.openssl.org:openssl/security.git`.
+
+Everything in this section is to be done as the `openssl` user on
+`dev.openssl.org`, so if you haven't done that yet, you now *must* perform
+the steps described in [SSH access](#ssh-access) above.
+
+Check that the release has been uploaded properly.  The release tarballs and
+associated files should be in `~openssl/dist/new`.  They should be owned by
+the `upload` userid and world-readable.
+
+Copy the tarballs to appropriate directories.  This can be done using the
+do-release.pl script.  See `$TOOLS/release-tools/DO-RELEASE.md` for a
+description of the options.  For example:
+
+    perl ~openssl/do-release.pl --copy --move
+
+This will copy the relevant files to the website and move them from
+`~openssl/dist/new` to `~openssl/dist/old` so they will not seen by a
+subsequent release.  Alternatively if you want to perform one release at a
+time or copy/move the files manually, see below.
+
+The `do-release.pl` script will display the commands you will need to issue
+to send the announcement emails later.  Keep a note of those commands for
+future reference.
+
+For public releases (including security releases), verify that the tarballs
+are available via FTP:
+
+    ls /srv/ftp/source
+
+For premium releases, verify that the tarballs are available via SFTP:
+
+    ls /srv/premium
+
+## on Github
```

Comment:
This step needs to be done AFTER the `Update source repositories` step otherwise the tags won't be present.

Reply:
I'd like to argue that they should *both* be done after the source update step.